### PR TITLE
feat: output cause on JS runtime errors

### LINKED ISSUE #12308: Output cause on JS runtime errors
Since Deno 1.13.0, Error Cause Proposal is supported, so the `Error` object can hold inner errors. However, their contents are usually not displayed and I need to explicitly access the `cause` property.

```js
function bar() {
  throw new Error("bar");
}

function foo() {
  try {
    bar();
  } catch (e) {
    throw new Error("foo", { cause: e });
  }
}

foo();
```

terminal

I think it should also show inner errors, just like Firefox does.

Firefox console

### PR DESCRIPTION
**Pull request by crowlKats**
Closes #12308

### CODE REVIEW

File: core/error.rs [hunk 1]
```diff
@@ -92,6 +92,7 @@ pub fn get_custom_error_class(error: &Error) -> Option<&'static str> {
 #[derive(Debug, PartialEq, Clone)]
 pub struct JsError {
   pub message: String,
+  pub cause: Box<Option<JsError>>,
```
Comment (bartlomieju) [inline core/error.rs:95]: Maybe `Option<Box<JsError>>`?

File: cli/fmt_errors.rs [hunk 2]
```diff
@@ -128,9 +128,11 @@ fn format_frame(frame: &JsStackFrame) -> String {
   result
 }
 
+#[allow(clippy::too_many_arguments)]
```
Comment (bartlomieju) [inline cli/fmt_errors.rs:131]: 😬 please add a TODO to refactor this function

**Comment by bartlomieju** [comment]
This looks mostly good to me. @nayeemrmn can you take a look?

File: core/error.rs [hunk 3]
```diff
@@ -180,53 +181,69 @@ impl JsError {
...
+        } else if !message_prop.is_empty() {
+          format!("Uncaught {}", message_prop)
+        } else {
+          "Uncaught".to_string()
+        };
+        let cause = cause.and_then(|cause| {
+          if cause.is_undefined() {
+            None
+          } else {
+            Some(Box::new(JsError::from_v8_exception(scope, cause)))
```
Comment (nayeemrmn) [inline core/error.rs:207]: Potential infinite for `error.cause = error`, may need a `seen: HashSet<_>` arg on this method. V8 inspector nicely cuts off cause printing where the loop is detected, as seen in `deno repl`: 
image
Firefox freaks out.

File: cli/fmt_errors.rs (hunk 2 above)
Comment (crowlKats) [reply cli/fmt_errors.rs:131]: eh, the function is called in a single place, and to resolve this we'd need to add a new struct, dont see too much value for that, imo we can leave as-is

**Review by bartlomieju — APPROVED** [review]
LGTM, nice work

**Comment by exe-dealer** [comment]
why this PR not included in 1.17.2 ? @crowlKats

**Comment by kitsonk** [comment]
New features are generally not included in patch release.  It will be included in 1.18.

**Pull request merged**